core/pta/bcm/wdt.c

```diff
@@ -0,0 +1,199 @@
+			       TEE_Param params[TEE_NUM_PARAMS] __unused)
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (exp_param_types != param_types) {
+		EMSG("Invalid Param types");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	wd_pd.chip.ops->start(&wd_pd.chip);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result pta_wd_ping(uint32_t param_types,
+			      TEE_Param params[TEE_NUM_PARAMS] __unused)
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (exp_param_types != param_types) {
+		EMSG("Invalid Param types");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	wd_pd.chip.ops->ping(&wd_pd.chip);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result pta_wd_stop(uint32_t param_types,
+			      TEE_Param params[TEE_NUM_PARAMS] __unused)
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (exp_param_types != param_types) {
+		EMSG("Invalid Param types");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	wd_pd.chip.ops->stop(&wd_pd.chip);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result pta_wd_set_timeout(uint32_t param_types,
+				     TEE_Param params[TEE_NUM_PARAMS])
+{
+	uint32_t timeout = 0;
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (exp_param_types != param_types) {
+		EMSG("Invalid Param types");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	timeout = params[0].value.a;
+
+	wd_pd.chip.ops->set_timeout(&wd_pd.chip, timeout);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result invoke_command(void *session_context __unused,
+				 uint32_t cmd_id,
+				 uint32_t param_types,
+				 TEE_Param params[TEE_NUM_PARAMS])
+{
+	TEE_Result res = TEE_SUCCESS;
+
+	DMSG("command entry point[%d] for \"%s\"", cmd_id, SEC_WD_TA_NAME);
+
+	switch (cmd_id) {
+	case PTA_BCM_SEC_WD_CMD_CONFIG:
+		res = pta_wd_config(param_types, params);
+		break;
+	case PTA_BCM_SEC_WD_CMD_START:
+		res = pta_wd_start(param_types, params);
+		break;
+	case PTA_BCM_SEC_WD_CMD_PING:
+		res = pta_wd_ping(param_types, params);
+		break;
+	case PTA_BCM_SEC_WD_CMD_STOP:
+		res = pta_wd_stop(param_types, params);
+		break;
+	case PTA_BCM_SEC_WD_CMD_SET_TIMEOUT:
+		res = pta_wd_set_timeout(param_types, params);
+		break;
+	default:
+		EMSG("cmd: %d Not supported %s", cmd_id, SEC_WD_TA_NAME);
+		res = TEE_ERROR_NOT_SUPPORTED;
+		break;
+	}
+
+	return res;
+}
+
+pseudo_ta_register(.uuid = SEC_WD_SERVICE_UUID,
+		   .name = SEC_WD_TA_NAME,
+		   .flags = PTA_DEFAULT_FLAGS,
+		   .create_entry_point = create_ta,
+		   .destroy_entry_point = destroy_ta,
+		   .open_session_entry_point = open_session,
+		   .close_session_entry_point = close_session,
```

Comment:
Since create/destroy/open/close do nothing more than print a debug trace, I recommend you drop them and provide only `.close_session_entry_point` here.

Reply:
Hi @jforissier 

Thanks for reviewing the changes.
I'd kept these APIs for completeness. Since all these apis(create/destroy/open/close) are not doing anything except for printing a msg. Shall i also remove close ? as i see event the `.close_session_entry_point` is also called conditionally in pseudo_ta.c file
`if (stc->pseudo_ta->close_session_entry_point)`
`    stc->pseudo_ta->close_session_entry_point(s->user_ctx);`

Reply:
I'm sorry I meant keep only `.invoke_command_entry_point` :open_mouth: